api/src/main/java/org/openmrs/module/fhir2/api/impl/FhirPatientServiceImpl.java

```diff
@@ -106,4 +107,26 @@ public IBundleProvider searchForPatients(StringAndListParam name, StringAndListP
 		
 		return searchQuery.getQueryResults(theParams, dao, translator, searchQueryInclude);
 	}
+	
+	@Override
+	@Transactional(readOnly = true)
+	public IBundleProvider getPatientEverything(TokenAndListParam patientId) {
```

Comment:
Why do we take a `TokenAndListParam` here? Aren't we always expected to have a single patient? (I get that we need to pass a `TokenAndListParam` as the search parameter; I just think we should construct the one-value `AndListParam` here instead of in the `ResourceProvider`... that way, the interface for the service method is clear that it *only* expects a single value).

Reply:
@ibacher Should we create and pass just a `TokenParam` from resource provider to service?

Reply:
Yes